cucumber/authenticators_jwt/features/authn_jwt_fetch_signing_key.feature

```diff
@@ -0,0 +1,535 @@
+    And I permit host "myapp" to "execute" it
+    And I add the secret value "test-secret" to the resource "cucumber:variable:test-variable"
+    And I issue a JWT token:
+    """
+    {
+      "host":"myapp",
+      "project-id": "myproject"
+    }
+    """
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the JWT token
+    Then the HTTP response status code is 401
+    And The following appears in the log after my savepoint:
+    """
+    CONJ00087E Failed to fetch JWKS from 'unknown-host.com'
+    """
+
+  Scenario: provider uri configured dynamically changed to jwks uri
+    Given I initialize JWKS endpoint with file "myJWKs.json"
+    And I load a policy:
+    """
+    - !policy
+      id: conjur/authn-jwt/keycloak
+      body:
+      - !webservice
+        annotations:
+          description: Authentication service for JWT tokens, based on keycloak JWKs.
+
+      - !variable
+        id: provider-uri
+
+      - !variable
+        id: token-app-property
+
+      - !variable
+        id: issuer
+
+      - !group hosts
+
+      - !permit
+        role: !group hosts
+        privilege: [ read, authenticate ]
+        resource: !webservice
+
+    - !host
+      id: myapp
+      annotations:
+        authn-jwt/keycloak/project-id: myproject
+
+    - !grant
+      role: !group conjur/authn-jwt/keycloak/hosts
+      member: !host myapp
+
+    - !host
+      id: alice
+      annotations:
+        authn-jwt/keycloak/email: alice@conjur.net
+
+    - !grant
+      role: !group conjur/authn-jwt/keycloak/hosts
+      member: !host alice
+    """
+    And I am the super-user
+    And I successfully set authn-jwt "provider-uri" variable with OIDC value from env var "PROVIDER_URI"
+    And I successfully set authn-jwt "token-app-property" variable with OIDC value from env var "ID_TOKEN_USER_PROPERTY"
+    And I successfully set authn-jwt "issuer" variable with OIDC value from env var "PROVIDER_ISSUER"
+    And I fetch an ID Token for username "alice" and password "alice"
+    And I save my place in the log file
+    When I authenticate via authn-jwt with the ID token
+    Then host "alice" has been authorized by Conjur
+    And The following lines appear in the log after my savepoint:
+      |                                                                                                 |
+      |CONJ00076I Selected signing key interface: 'provider-uri'                                        |
+      |CONJ00072I Fetching JWKS from 'https://keycloak:8443/auth/realms/master'...                      |
+      |CONJ00054I "issuer" value will be taken from 'cucumber:variable:conjur/authn-jwt/keycloak/issuer'|
+      |CONJ00055I Retrieved "issuer" with value 'http://keycloak:8080/auth/realms/master'               |
+      |CONJ00098I Successfully found JWT identity 'host/alice'                                          |
+    And I update the policy with:
```

Comment:
Single test should be
Given
When
Then

These three shouldnt repeat themselves.

It should be
Given i connect with provider URI successfuly
When i change to jwks-uri
Then i Connect with it also successfuly

So change here to When and all above to And